spec.md

```diff
@@ -792,133 +727,44 @@ In the following example, all variants have the type `UInt<0>`{.firrtl}.
 {|a, b, c|}
 ```
 
-## Analog Type
-
-The analog type specifies that a wire or port can be attached to multiple drivers.
-`Analog`{.firrtl} cannot be used as part of the type of a node or register, nor can it be used as part of the datatype of a memory.
-In this respect, it is similar to how `inout`{.firrtl} ports are used in Verilog, and FIRRTL analog signals are often used to interface with external Verilog or VHDL IP.
-
-In contrast with all other ground types, analog signals cannot appear on either side of a connection statement.
-Instead, analog signals are attached to each other with the commutative `attach`{.firrtl} statement.
-An analog signal may appear in any number of attach statements, and a legal circuit may also contain analog signals that are never attached.
-The only primitive operations that may be applied to analog signals are casts to other signal types.
-
-When an analog signal appears as a field of an aggregate type, the aggregate cannot appear in a standard connection statement.
-
-As with integer types, an analog type can represent a multi-bit signal.
-When analog signals are not given a concrete width, their widths are inferred according to a highly restrictive width inference rule, which requires that the widths of all arguments to a given attach operation be identical.
-
-``` firrtl
-Analog<1>  ; 1-bit analog type
-Analog<32> ; 32-bit analog type
-Analog     ; analog type with inferred width
-```
-
-## Reference Types
-
-References can be exported from a module for indirect access elsewhere, and are captured using values of reference type.
-
-For use in cross-module references (hierarchical references in Verilog), a reference to a probe of the circuit component is used.
-See [@sec:probes] for details.
-
-Using probe-type ports, modules may expose internals for reading and forcing without routing wires out of the design.
+## Probe Types
 
-This is often useful for testing and verification, where probe types allow reads of the entities to be explicitly exported without hard-coding their place in the design.
-Instead, by using probe-type references, a testbench module may express accesses to the internals which will resolve to the appropriate target language construct by the compiler (e.g., hierarchical reference).
+Probe types expose and provide access to circuit components contained inside a module.
 
-Reference ports are not expected to be synthesizable or representable in the target language and are omitted in the compiled design; they only exist at the FIRRTL level.
+Probe types are useful for testing and verification, since they allow a design to read to and write from circuit components without knowing the hierarchical path to that component in the design.
```

Comment:
this seems like a possible mischaracterization... in firrtl there is no concept of hierarchical path to a component in a design (except in targets...), in verilog there is. We could have just implemented probes like that in the firrtl spec, the advantage isn't really that you do or do not know the hierarchical path to the component in the design.  I'm not sure that is what makes them useful for testing and verification, they are useful in verilog because there is a sense that the verilog hierarchy is "real" and XMRs are "not real" -- you can add XMRs without adding any "real" ports. Would love @seldridge, @darthscsi  to weigh in on this rewrite.

Reply:
A major use case of probes is that you can use them without knowing the hierarchical path which makes them work for separate compilation where annotations are intractable. They are highly amenable to compiler implementations because they work with hierarchy manipulations and optimizations (e.g., constant propagation) whereas annotations were a hug pain here.

That said, drawing a comparison to something which doesn't exist in FIRRTL (a hierarchical path) may not be the right way to explain this. I.e., they are useful for verification because they let you read from circuit components without creating ports. (This is restating Megan's penultimate sentence.)

Reply:
Removed the references to hierarchical paths.

I did slip one usage of the term "physical hardware". Please consider this an informal statement. We will work towards a formalization of what this means before too long.